src/crewai/agents/agent_interface.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+from typing import Optional, List, Any, Dict
+
+from pydantic import BaseModel, PrivateAttr, Field
+
+from crewai.utilities import I18N
+
+
+class AgentWrapperParent(ABC, BaseModel):
+    _i18n: I18N = PrivateAttr(default=I18N())
+    data: Dict[str, Any] = Field(
+        default_factory=dict,
+        description="Data storage for children, as pydantic doesn't play well with inheritance.",
+    )
+    role: str = Field(description="Role of the agent", default="")
+    allow_delegation: bool = Field(
+        description="Allow delegation of tasks to other agents?", default=False
+    )
+
+    @property
+    def i18n(self) -> I18N:
+        if hasattr(self, "_agent") and hasattr(self._agent, "i18n"):
```

Comment:
Could we prevent this kind of indirection by using some sort of interface? We are moving fast to become a strongly-typed library and this makes it very hard to make sure things are working properly.

Reply:
If an offician AgentInterface existed, this could be part of it :)